runtimes/pythonrt/runner_manager_docker.go

```diff
@@ -0,0 +1,162 @@
+package pythonrt
+
+import (
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type DockerRuntimeConfig struct {
+	WorkerAddressProvider func() string
+	LogRunnerCode         bool
+	LogBuildCode          bool
+}
+
+type dockerRunnerManager struct {
+	logger                *zap.Logger
+	client                *dockerClient
+	runnerIDToContainerID map[string]string
+	mu                    *sync.Mutex
+	workerAddressProvider func() string
+}
+
+func (c DockerRuntimeConfig) validate() error {
+	return nil
+}
+
+func configureDockerRunnerManager(log *zap.Logger, cfg DockerRuntimeConfig) error {
+	if err := cfg.validate(); err != nil {
+		return err
+	}
+
+	dc, err := NewDockerClient(log, cfg.LogRunnerCode, cfg.LogBuildCode)
+	if err != nil {
+		return err
+	}
+
+	_, err = dc.ensureNetwork()
+	if err != nil {
+		return err
+	}
+
+	log.Info(fmt.Sprintf("docker connected and synced succesffully, there are %d active runners on network %s", dc.ActiveRunnersCount(), networkName))
+
+	if len(dc.activeRunnerIDs) != 0 {
```

Comment:
`> 0`

Also comment on the reason you're doing this.

Reply:
done